pkg/daemon/daemon.go

```diff
@@ -256,15 +270,51 @@ func cmdRun(p *ptpProcess) {
 
 	done := make(chan struct{})
 
-	scanner := bufio.NewScanner(cmdReader)
-	go func() {
-		for scanner.Scan() {
-			output := scanner.Text()
-			fmt.Printf("%s\n", output)
-			extractMetrics(p.name, p.iface, output)
-		}
-		done <- struct{}{}
-	}()
+	if !stdoutToSocket {
+		scanner := bufio.NewScanner(cmdReader)
+		go func() {
+			for scanner.Scan() {
+				output := scanner.Text()
+				fmt.Printf("%s\n", output)
+				extractMetrics(p.name, p.iface, output)
+			}
+			done <- struct{}{}
+		}()
+	} else {
+		go func(io.ReadCloser) {
+			for {
+			connect:
+				select {
+				case <-p.exitCh:
+					done <- struct{}{}
+				default:
+					c, err = net.Dial("unix", "/cloud-native/events.sock")
```

Comment:
please reuse the function you already have `handleConnection`

Reply:
i delete that function

Reply:
function is not required ,since the code is  used only  here